pkg/runner/run_context.go

```diff
@@ -118,7 +118,7 @@ func (rc *RunContext) startJobContainer() common.Executor {
 			rc.JobContainer.Remove().IfBool(!rc.Config.ReuseContainers),
 			rc.JobContainer.Create(),
 			rc.JobContainer.Start(false),
-			rc.JobContainer.CopyDir(copyToPath, rc.Config.Workdir+"/.").IfBool(copyWorkspace),
+			rc.JobContainer.CopyDir(copyToPath, rc.Config.Workdir+"/.", true).IfBool(copyWorkspace),
```

Comment:
Instead of hard coding `true` here (and `false` in `pkg/runner/step_context.go#L269), could you use a command line switch do apply/ignore `.gitignore` patterns? This would fix issue #325 as well.